docs_2.0/manual_EN/5.deployment-and-installation/1.resource-preparations.md

```diff
@@ -51,6 +51,102 @@ You must have the correct version of the software listed below to compile Nebula
 | libstdc++-static | Any stable version | Only needed in CentOS 8+, RedHat 8+, and Fedora systems.  |
 | libasan          | Any stable version | Only needed in CentOS 8+, RedHat 8+, and Fedora systems.  |
 
+Other third-party software will be automatically downloaded and installed to the build directory at the configure (cmake) stage.
+
+### Prepare software for compiling Nebula Graph
+
+This section guides you through the downloading and installation of software required for compiling Nebula Graph.
+
+1. Install dependencies.
+
+   * For CentOS, RedHat, and Fedora users, run the following commands.
+
+        ```bash
+        $ yum update
+        $ yum install -y make \
+                         m4 \
+                         git \
+                         wget \
+                         unzip \
+                         xz \
+                         readline-devel \
+                         ncurses-devel \
+                         zlib-devel \
+                         gcc \
+                         gcc-c++ \
+                         cmake \
+                         gettext \
+                         curl \
+                         redhat-lsb-core
+        # For CentOS 8+, RedHat 8+, and Fedora, install libstdc++-static, libasan as well
+        $ yum install -y libstdc++-static libasan
+        ```
+
+   * For Debian and Ubuntu users, run the following commands.
+
+        ```bash
+        $ apt-get update
+        $ apt-get install -y make \
+                             m4 \
+                             git \
+                             wget \
+                             unzip \
+                             xz-utils \
+                             curl \
+                             lsb-core \
+                             build-essential \
+                             libreadline-dev \
+                             ncurses-dev \
+                             cmake \
+                             gettext
+        ```
+
+2. Check if the GCC and cmake on your host are in the right version. See [Software requirements for compiling Nebula Graph](#software-requirements-for-compiling-nebula-graph) for the required versions.
+
+    ```bash
+    $ g++ --version
+    $ cmake --version
+    ```
+
+    If your GCC and cmake are in the right version, then you are all set. If they are not, follow the sub-steps as follows.
+
+    1. Clone the nebula-common repository to your host.
+
+        ```bash
+        $ git clone https://github.com/vesoft-inc/nebula-common.git
```

Comment:
for 2.0-alpha, beta, etc, use `git clone -b 2.0-alpha` instead of cloning the master branch.